interp/interp.go

```diff
@@ -77,6 +77,16 @@ type Runner struct {
 	Context context.Context
 
 	stopOnCmdErr bool // set -e
+
+	// KillTimeout holds how much time the interpreter wait until it force-kill
+	// an process, if it didn't self-exit after receive an Interrupt signal.
```

Comment:
KillTimeout holds how much time the interpreter will wait for a program to stop after being sent an interrupt signal, after which a kill signal will be sent. This process will happen when the interpreter's context is cancelled.